databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    kashooEntityType: {
```

Comment:
I'm now questioning whether we want to even have this association stored here, and that this document really just stores the raw data as received from the external source.  It makes more sense to me that a Kashoo entity only gets associated with raw data when we decide to do something with it, which will likely include a bunch of additional metadata, and some of which you've alluded to in comments below (processed timestamp, etc).  Upon that operation happening, we should probably store a new document including the associations and relevant metadata.  My thinking is to keep document types as specific and simple as possible, with only a single purpose (in this case, simply a record of when/what external data was received)